manifests/plugin/remote_execution/ssh.pp

```diff
@@ -64,6 +68,22 @@
       cwd     => $ssh_identity_dir,
       creates => $ssh_identity_path,
     }
+    if $install_key {
+      # Ensure the .ssh directory exists with the right permissions
+      file { "/root/.ssh":
+        ensure  => directory,
+        owner   => 'root',
+        group   => 'root',
+        mode    => '0700',
+      }
+
+      exec { 'install_ssh_key':
+        path    => '/usr/bin:/usr/sbin:/bin',
+        command => "cat ${ssh_identity_path}.pub >> /root/.ssh/authorized_keys",
+        unless  => "grep '`cat ${ssh_identity_path}.pub`' /root/.ssh/authorized_keys",
```

Comment:
Actually there's a built-in type for authorized keys

https://docs.puppet.com/puppet/latest/types/ssh_authorized_key.html